### PR TITLE
Attempt to fix #10610

### LINKED ISSUE #10610: limit(3**n*3**(-n - 1)*(n + 1)**2/n**2, n, oo) is wrong
```
In [41]: limit(3**n*3**(-n - 1)*(n + 1)**2/n**2, n, oo)
Out[41]: 1
```

From https://stackoverflow.com/questions/35428617/sympy-fails-limit-test. If you expand the expression first the terms cancel and it gets the answer right. The answer should be 1/3.

### PR DESCRIPTION
**Pull request by Shikhar1998**
An attempt to fix the issue #10610 using ratsimp method.
The limits uses simplifiy technique to reduce the given function but in some cases this simplication is not absolute like here so by using ratsimp I have tried enhancing the extent of simplification of the given function.
```ratsimp(3**n*3**(-n - 1)*(n + 1)**2/n**2)```
returns ```1/3 + (2*n + 1)/(3*n**2)```
which by ```inve = e.subs(z, 1/u)```
is further reduced.
Hope it corrects the problem..

### CODE REVIEW

**Comment by Shikhar1998** [comment]
@asmeurer can u please review it (It seems to me that using ratsimp makes the code much slower but while testing it on the local machine it passes all cases) .

**Comment by Shikhar1998** [comment]
I have added selectivity on the functions that would be simplified.
@asmeurer can u please review it.

File: sympy/series/limits.py [hunk 1]
```diff
@@ -88,10 +88,15 @@ class Limit(Expr):
     """
 
     def __new__(cls, e, z, z0, dir="+"):
+        temp = e
         e = sympify(e)
+        try:
+            if e == temp:
+                e = ratsimp(e)
+        except Exception :
```
Comment (asmeurer) [inline sympy/series/limits.py]: Please don't add a generic exception handler. What is the specific exception you are trying to catch here?

**Comment by asmeurer** [comment]
Sorry, but the fix for this issue should go in the limit algorithm, not in the `Limit` constructor.

**Comment by Shikhar1998** [comment]
@asmeurer ,Thanks for the response . 
I checked through the code again . The Gruntz algorithm does give the correct answer (1/3) but the problem here was in the application of the leading term logic which should be applied only on *polynomials*. The conditions were a bit loose. I have tried to correct them. I think there is no need to do use ratsimp as Gruntz does give the correct answer. Please check.

**Comment by jksuom** [comment]
It looks like `bin/test test_limits` is failing.

**Comment by Shikhar1998** [comment]
@jksuom It says time exceeded for the output but I can't understand the reason because I did not change the complexity much .Moreover, I also don't know on why test cases it is failing . I tried it on my local machine it does passes all test cases.

**Comment by jksuom** [comment]
A test may fail on Travis if it takes a long time even if would eventually pass. You can find such tests locally by running bin/test with the option -v.

**Pull request closed without merging**